internal/gmm/message/build.go

```diff
@@ -789,12 +789,23 @@ func BuildConfigurationUpdateCommand(ue *context.AmfUe, anType models.AccessType
 		configurationUpdateCommand.ShortNameForNetwork.SetIei(nasMessage.ConfigurationUpdateCommandShortNameForNetworkType)
 	}
 
+	now := time.Now()
+	universalTimeAndLocalTimeZone := nasConvert.UniversalTimeAndLocalTimeZoneToNas(now)
+	universalTimeAndLocalTimeZone.SetIei(nasMessage.ConfigurationUpdateCommandUniversalTimeAndLocalTimeZoneType)
+	configurationUpdateCommand.UniversalTimeAndLocalTimeZone = &universalTimeAndLocalTimeZone
+
 	if ue.TimeZone != "" {
```

Comment:
According to 24.501, 8.2.19.9 Local time zone ,
> This IE may be included to assign a new local time zone to the UE.

When is the ue.TimeZone set? Shall we make it like only when `ue.TimeZone != amfSelf.TimeZone` will AMF send configuration update command to update ue's local time zone?
(I'm not quite familiar with the time zone handling between ue and amf, so feel free to discuss if you have different understanding of spec.)

Reply:
Actually, I could not find the code that assigns the time zone value to ue.TimeZone, nor could I find the procedure through which the UE sends its time zone information to the network.
In addition, according to TS 22.042, it seems that providing the time information is the responsibility of the network.
Therefore, I think AMF should store the time zone information of the network.

```
if ue.TimeZone != amfSelf.TimeZone {
    ue.TimeZone = amfSelf.TimeZone
    // ...
    /* Set up the time zone part of the configuration update command */
}
```
Is this better?

Reply:
Sorry for my late reply. Yes, I think it is better. Thanks.